Save analysis area

henpe071 opened the pull request:
If an analysis is done within an interval or a bounding box , the interval and box are saved and can be showed by turning them on in the contex menu of the analysis. 

When an analysis is done it will check if an analysis with that name already exist in the target folder and if so add a number to the end so it doesn't overwrite.

Fixed so the analysis is done on the correct area when video is zoomed in.

Analysis interval, the one that shows where the analysis is done, is now a seperate interval. Intervals are now saved as pairs instead of analysisInterval.

henpe071 committed:
Save analysis area and don't override save data

If an analaysis is done within an interval or a bounding box they are saved and can be accessed by the context menu of the analysis.

when an analysis is done it will now check if another analysis with the same name already is saved and add a number to the end if that's the case to prevent overwriting.

henpe071 committed:
the analysis area now scales correctly when zoomed in

Also fixed the correct pois show when clicking an analysis directly

henpe071 committed:
bounding box and analysis interval now saves and load correctly

the analysis interval is now a seperate interval from the normal interval. Intervals now saves as pairs instead of AnalysisIntervals.
TODO Crash when changing analysis and clean up.

henpe071 committed:
Cleaning and fixed a crash when changing video

henpe071 committed:
Cleaned some things and added some comments

fredrikiselius commented on ViAn/Analysis/analysismethod.h:
```diff
@@ -35,7 +35,8 @@ class AnalysisMethod : public QObject ,public QRunnable{
 
 protected:
-    AnalysisInterval interval;
+    std::pair<int, int> interval;
```
Why not use the AnalysisInterval? It contains the same information

henpe071 replied:
I felt like i didn't need/want the extra things in the AnalysisInterval. The interval also isn't an AnalysisInterval in the way we used it before. And lastly, before the interval was sometimes a pair and sometimes an AnalysisInterval which wasn't really necessary. Now it's the same type all the time.

Both works, I just chose a pair.

fredrikiselius commented on ViAn/GUI/Analysis/analysisslider.cpp:
```diff
@@ -52,20 +52,39 @@ void AnalysisSlider::paintEvent(QPaintEvent *ev) {
         }
     }
 
+    // Draws the analysis interval
```
Is this the same code as for the interval drawing? If so, it should be reused

fredrikiselius commented on ViAn/GUI/Analysis/analysisslider.cpp:
```diff
@@ -52,20 +52,39 @@ void AnalysisSlider::paintEvent(QPaintEvent *ev) {
+        brush = Qt::darkMagenta;
+        double first = (groove_rect.left()+(double)m_ana_interval.first*c);
+        double second = (groove_rect.left()+(double)m_ana_interval.second*c);
+        if (m_ana_interval.first != -1 && m_ana_interval.second != -1 && m_ana_interval.first <= m_ana_interval.second) {
```
It quite hard to read the code with all these conditions that doesn't really say anything. 

 m_ana_interval.first != -1 && m_ana_interval.second != -1

Could for example be refactored into a "valid_interval" function or similar.

henpe071 committed:
Structured the code better in the analysis slider and added details to view menu

Structured the code a little better according to comments.
Added show/hide analysis details to the view menu and the setting persist when changing video or analysis.

henpe071 commented:
Structured the code a little better in the analysisslider and added the show/hide details in the view menu. The show/hide details now also remains even when changing video or analysis.

henpe071 committed:
Merge branch 'development' into Save-analysis-area

henpe071 commented:
TODO fix crash

henpe071 committed:
Remove some things

henpe071 committed:
Fix crash

Fuuuuuuuu

henpe071 committed:
Merge branch 'development' into Save-analysis-area
